### PR TITLE
fix: mitigate user settings race conditions

### PR DESCRIPTION
**Pull request by ashtonG**
## Description

This mitigates an issue that appeaers on installations that have rbac enabled. When updating settings, it's sometimes possible that a race condition happens between the update request and the polled settings request. becaue we optimistically update the local state, this can cause the app to appear like it's not accepting the changes, only for it to eventually settle after the next polling call.

We fix this issue by adding a grace period to the beginning of the polling method. we keep track of all in-flight update calls and wait for up to 500ms for those to complete before making the `getUserSetting` call. If there are no updates in flight, the poll continues without waiting.

## Test Plan

workspace filters should be preserved on servers with rbac enabled.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

should mitigate/fix [WEB-1666]

[WEB-1666]: https://hpe-aiatscale.atlassian.net/browse/WEB-1666?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hkang1 — COMMENTED** [review]
overall looks good, one question about possibly #updates filling up with lots of broken promises :)

**Comment by ClaireNeveu** [comment]
Is this applicable to all polling stores or just user settings?

File: webui/react/src/stores/userSettings.tsx [hunk 1]
```diff
@@ -302,15 +308,21 @@ export class UserSettingsStore extends PollingStore {
...
-        publicMessage: `Unable to update user settings for key: ${key}.`,
-        publicSubject: 'Some POST user settings failed.',
-        silent: true,
-        type: ErrorType.Api,
-      }),
-    );
+    const promise = updateUserSetting({ settings: dbUpdates })
+      .then(() => {
+        this.#updates = this.#updates.filter((p) => p !== promise);
+      })
```
[the remark this answers is not in the document]
Comment (ashtonG) [reply webui/react/src/stores/userSettings.tsx]: good looking out -- updated this to remove the promise no matter what.

**Comment by ashtonG** [comment]
@ClaireNeveu only other possibly affected store is the user store. All other stores are either read only, or do not update the client state optimistically. While the code that uses the user store is not susceptible to the race condition, i've pushed an update to ensure that it doesn't happen in the future.

**Pull request merged**